chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -264,6 +264,23 @@ def self.unclaimed(node, include_mounted=false)
           end
         end
 
+        # return the disks that are unclaimed and prepared for given role (or with no role)
+        # (sort the result so the ones with exact role come first)
```

Comment:
@jsuchome your last proposal kind of makes sense.

And just to clarify, the reason I don't like the current way is because the "take first" vs "take all" choice is actually something we should move away from, especially with your changes that allow something much more fine-grained.